[BUG] delta_lake_write_test.py::test_delta_write_optimized_supported_types_partitioned FAILED ON "Part of the plan is not columnar class org.apache.spark.sql.execution.command.ExecutedCommandExec"

**Describe the bug**

delta_lake_write_test.py::test_delta_write_optimized_supported_types_partitioned FAILED ON "Part of the plan is not columnar class org.apache.spark.sql.execution.command.ExecutedCommandExec"

First reported from CI integration/1325 on the spark320 shim

```
 # JVM exception message.
 >               raise converted from None
 E               pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.command.ExecutedCommandExec
[2025-07-18T09:50:14.230Z] E               Execute SaveIntoDataSourceCommand
 E                  +- SaveIntoDataSourceCommand org.apache.spark.sql.delta.sources.DeltaDataSource@510022ec, Map(__partition_columns -> ["c0"], path -> /tmp/pyspark_tests//it-test-320-cuda12-jenkins-rapids-integration-dev-github--66bl2-gw2-1410340-467129944//DELTA_DATA/GPU), ErrorIfExists
 E                        +- LogicalRDD [c0#133653, c1#133654, c2#133655, c3#133656, c4#133657, c5#133658L, c6#133659, c7#133660, c8#133661, c9#133662, c10#133663, c11#133664, c12#133665, c13#133666, c14#133667, c15#133668, c16#133669, c17#133670, c18#133671, c19#133672, c20#133673, c21#133674, c22#133675, c23#133676, ... 40 more fields], false
```


**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/src/main/python/delta_lake_write_test.py
+++ b/integration_tests/src/main/python/delta_lake_write_test.py
@@ -989,7 +989,7 @@ def test_delta_write_optimized_supported_types(spark_tmp_path):
 @delta_lake
 @ignore_order(local=True)
 @pytest.mark.skipif(is_before_spark_320(), reason="Delta Lake writes are not supported before Spark 3.2.x")
-@pytest.mark.skipif(not is_databricks_runtime() and is_spark_353_or_later(), reason="Delta Lake optimized writes are not supported before Spark 3.5.3 on Apache Spark")
+@pytest.mark.skipif(not is_databricks_runtime() and is_before_spark_353(), reason="Delta Lake optimized writes are not supported before Spark 3.5.3 on Apache Spark")
 def test_delta_write_optimized_supported_types_partitioned(spark_tmp_path):
     data_path = spark_tmp_path + "/DELTA_DATA"
     confs=copy_and_update(writer_confs, delta_writes_enabled_conf, {

